agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +456,168 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
 func (cs *CredentialSpecResource) UnmarshallPlatformSpecificFields(credentialSpecResourceJSON CredentialSpecResourceJSON) {
 	return
 }
+
+// This function makes a copy of the local credential spec file to avoid modifying customer resources
+// Example Usage:
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-cred.json") - single file
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-dir\gmsa-cred.json") - Relative path with Windows file path separators
+func (cs *CredentialSpecResource) copyCredentialSpecLocalFile(credentialspec string) (string, error) {
```

Comment:
I don't see any usage for this private function in this PR. Where are we planning to use it?

Reply:
old code, removed in next PR